src/coreclr/jit/llvmlssa.cpp

```diff
@@ -301,80 +525,369 @@ class ShadowStackAllocator
+                VarSetOps::RemoveElemD(m_compiler, m_liveDefs, varDsc->lvVarIndex);
+            }
+            else if (isBorn)
+            {
+                VarSetOps::AddElemD(m_compiler, m_liveDefs, varDsc->lvVarIndex);
+            }
+
+            JITDUMPEXEC(PrintCurrentLiveCandidates());
+        }
+
+        void SetLastActiveUseIsLastUse(LclSsaVarDsc* ssaDsc)
+        {
+            assert(!IsLastActiveUseLastUse(ssaDsc));
+            *GetRawActiveUseCountRef(ssaDsc) |= LAST_ACTIVE_USE_IS_LAST_USE_BIT;
+        }
+
+        bool IsLastActiveUseLastUse(LclSsaVarDsc* ssaDsc)
+        {
+            return (*GetRawActiveUseCountRef(ssaDsc) & LAST_ACTIVE_USE_IS_LAST_USE_BIT) != 0;
+        }
+
+        void ResetActiveUseCount(LclSsaVarDsc* ssaDsc)
+        {
+            unsigned* pCount = GetRawActiveUseCountRef(ssaDsc);
+            assert((*pCount == ValueNumStore::NoVN) || (GetActiveUseCount(ssaDsc) <= 1));
+            *pCount = 0;
+        }
+
+        void IncrementActiveUseCount(LclSsaVarDsc* ssaDsc)
+        {
+            assert(GetActiveUseCount(ssaDsc) < ssaDsc->GetNumUses());
+            (*GetRawActiveUseCountRef(ssaDsc))++;
+        }
+
+        void DecrementActiveUseCount(LclSsaVarDsc* ssaDsc)
+        {
+            assert(GetActiveUseCount(ssaDsc) > 0);
+            (*GetRawActiveUseCountRef(ssaDsc))--;
+        }
+
+        unsigned GetActiveUseCount(LclSsaVarDsc* ssaDsc)
+        {
+            return *GetRawActiveUseCountRef(ssaDsc) & ~LAST_ACTIVE_USE_IS_LAST_USE_BIT;
+        }
+
+        unsigned* GetRawActiveUseCountRef(LclSsaVarDsc* ssaDsc)
+        {
+            // Another little hack for storing per-def information, needed for last-use detection.
+            return ssaDsc->m_vnPair.GetConservativeAddr();
+        }
+
+        bool IsCandidateLocal(LclVarDsc* varDsc)
+        {
+            return (varDsc->GetRegNum() == REG_STK_CANDIDATE_TENTATIVE) ||
+                   (varDsc->GetRegNum() == REG_STK_CANDIDATE_COMMITED);
+        }
+
+#ifdef DEBUG
+        void VerifyActiveUseCounts()
         {
-            LclVarDsc* varDsc = m_compiler->lvaGetDesc(lclNum);
-            if (varDsc->TypeGet() == TYP_REF)
-            {
-                m_spillLclsRef.Push(lclNum);
-            }
-            else if (varDsc->TypeGet() == TYP_BYREF)
+            for (unsigned lclNum = 0; lclNum < m_compiler->lvaCount; lclNum++)
             {
-                m_spillLclsByref.Push(lclNum);
+                LclVarDsc* varDsc = m_compiler->lvaGetDesc(lclNum);
+                if (IsCandidateLocal(varDsc))
+                {
+                    unsigned defCount = varDsc->lvPerSsaData.GetCount();
+                    for (unsigned defIndex = 0; defIndex < defCount; defIndex++)
+                    {
+                        // We should have seen the exact totality of all uses in the IR walk. Make an exemption
+                        // for an unused implicit definition, of which the use count we do not uninitialize.
```

Comment:
nit - English reads a little funny.
```suggestion
                        // for an unused implicit definition, for which we do not uninitialize the use count .
```